### PR TITLE
Add RAJA Features to Caliper Performance Metrics

### PR DESCRIPTION
**Pull request by michaelmckinsey1**
# Add RAJA Features to Caliper Performance Metrics

- This PR is a feature
- It does the following:
  - Modifies the Caliper instrumentation to include per-kernel RAJA features in the performance metrics

### CODE REVIEW

**Review by pearce8 — APPROVED** [review]

**Comment by pearce8** [comment]
@MrBurmark This PR adds the feature info to metadata in Caliper output we discussed last week.  Please review/merge.

File: src/common/KernelBase.cpp [hunk 1]
```diff
@@ -77,6 +77,13 @@ KernelBase::KernelBase(KernelID kid, const RunParams& params)
                                            CALI_ATTR_ASVALUE |
                                            CALI_ATTR_AGGREGATABLE |
                                            CALI_ATTR_SKIP_EVENTS);
+  for (unsigned i = 0; i < FeatureID::NumFeatures; ++i) {
+    std::string feature = getFeatureName(static_cast<FeatureID>(i));
```
Comment (MrBurmark) [inline src/common/KernelBase.cpp:81]: Can you pull out fid into a var to be consistent with the coding below?

File: src/common/KernelBase.hpp [hunk 2]
```diff
@@ -570,6 +570,7 @@ class KernelBase
   cali_id_t Bytes_Rep_attr;
   cali_id_t Flops_Rep_attr;
   cali_id_t BlockSize_attr;
+  std::map<std::string, cali_id_t> Features;
```
Comment (MrBurmark) [inline src/common/KernelBase.hpp:573]: How about adding `_attrs` to this variable name?

File: src/common/KernelBase.cpp (hunk 1 above)
Comment (michaelmckinsey1) [reply src/common/KernelBase.cpp]: Done. [cabcbb3](https://github.com/LLNL/RAJAPerf/pull/456/commits/cabcbb3afedad19725e63014b4ed4b8be686f521)

File: src/common/KernelBase.hpp (hunk 2 above)
Comment (michaelmckinsey1) [reply src/common/KernelBase.hpp]: Done [07baca4](https://github.com/LLNL/RAJAPerf/pull/456/commits/07baca41feeaf4040b9c047e1523d02ac075d1cb)

**Review by MrBurmark — APPROVED** [review]

**Pull request merged**